### PR TITLE
Adding docker workflow

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

Adding docker workflow

### Changes

adding docker workflow in ci-cd.yml

### Related Issues

[List of related issues](https://github.com/threefoldtech/grid_weblets/issues/947)

### CODE REVIEW

File: .github/workflows/ci-cd-docker.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: Build and push a new docker image based on project release tag.
+on:
+  release:
+    types: [published]
```
Comment (AhmedHanafy725) [inline .github/workflows/ci-cd-docker.yml:4]: this workflow should run also on pushing on the development branch

File: .github/workflows/ci-cd-docker.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: ${{ secrets.DOCKER_USERNAME }}/grid_weblets:${{ steps.previoustag.outputs.tag }}
+          build-args: VERSION=${VERSION},NETWORK=${NETWORK}
```
Comment (AhmedHanafy725) [inline .github/workflows/ci-cd-docker.yml:66]: not sure if the version and network vars will be still known in this step, can you please make sure that the resulted image contains the correct version and network?

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # This workflow will do a clean install of node dependencies, cache/restore them, build the source code and deploy to staging server by Copying the artifacts using ssh to play.dev.grid.tf
 
-name: CI/CD pipeline 
+name: CI/CD pipeline
```
Comment (AhmedHanafy725) [inline .github/workflows/ci-cd.yml:3]: can we please remove any changes in this file?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-cd.yml (hunk 3 above)
Comment (zaelgohary) [reply .github/workflows/ci-cd.yml:3]: done.

File: .github/workflows/ci-cd-docker.yml (hunk 1 above)
Comment (zaelgohary) [reply .github/workflows/ci-cd-docker.yml:7]: done.

File: .github/workflows/ci-cd-docker.yml (hunk 2 above)
Comment (zaelgohary) [reply .github/workflows/ci-cd-docker.yml:69]: Tested on a private repo and this is while releasing the master branch:
image

while pushing on development:
image

Comment (zaelgohary) [reply .github/workflows/ci-cd-docker.yml]: fixed.

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**